test/test_rospy/test/rostest/test_basic_services.py

```diff
@@ -213,6 +214,28 @@ def test_String_String(self):
             resp_req_kwds = self._test_req_kwds(name, Cls, {'str': String('FOO'), 'str2': Val('bar')})
             for resp in [resp_req, resp_req_naked, resp_req_kwds]:
                 self.assertEquals('FOObar', resp.str.data)
+
+    def test_String_String_unicode(self):
+        from std_msgs.msg import String
+        from test_rospy.srv import StringString, StringStringRequest
+        from test_rospy.msg import Val
+        import sys
+        Cls = StringString
+        Req = StringStringRequest
+
+        for name in [STRING_CAT_SERVICE_NAKED, STRING_CAT_SERVICE_WRAPPED]:
+            if sys.version < '3':
```

Comment:
@dirk-thomas changed to use sys.version_info.major
@mikepurvis as far as I remember, interpreting unicode has been changed between python 2 vs python 3
- http://lucumr.pocoo.org/2014/1/5/unicode-in-2-and-3/
- https://docs.python.org/3/howto/unicode.html#python-s-unicode-support
- https://docs.python.org/2/howto/unicode.html#python-2-x-s-unicode-support

```
k-okada@p40-yoga:/tmp/ros_comm/test/test_rospy/test/rostest$ python
Python 2.7.6 (default, Oct 26 2016, 20:30:19)
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> u'オープンソース' == 'オープンソース'
__main__:1: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them\
 as being unequal
False
```
and
```
k-okada@p40-yoga:~$ python3
Python 3.4.3 (default, Nov 17 2016, 01:08:31) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> u'オープンソース' == 'オープンソース'
True
```